backend/routes/user_routes.py

```diff
@@ -25,7 +27,7 @@ async def send_login_link(
     user.login_link = str(login_link_uuid)
     session.commit()
     # TODO: Implement this function to send WhatsApp messages
-    send_whatsapp_message(phone_number, user.login_link)
+    await send_whatsapp_message(phone_number, user.login_link)
```

Comment:
The phone number format needs to be validated to be globalised E.164 format. We can argue about whether it should be numeric at the SQL level or not, but on the wire it needs to be E.164 with a leading + sign.

In general, all data coming in should have validation.